src/Analyzers/CSharp/Analyzers/NamespaceMatchFolder/CSharpNamespaceMatchFolderDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Analyzers.NamespaceSync;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.NamespaceMatchFolder
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpNamespaceMatchFolderDiagnosticAnalyzer : AbstractNamespaceMatchFolderDiagnosticAnalyzer<NamespaceDeclarationSyntax>
+    {
+        protected override SyntaxNode GetNameSyntax(NamespaceDeclarationSyntax namespaceDeclaration) => namespaceDeclaration.Name;
```

Comment:
this would be a good sytnax fact.  I'ts something common to both VB and C# and is specifically about an exact sytnax-shape/relationship.  this is what ISyntaxFacts is actually for :)